src/Persistence/Sql.php

```diff
@@ -153,7 +153,12 @@ static function ($matches) use ($model, &$arguments) {
 
                 $identifier = substr($matches[0], 1, -1);
                 if ($identifier !== '' && !isset($arguments[$identifier])) {
-                    $arguments[$identifier] = $model->getField($identifier);
+                    // TODO horrible hack to render the field with a table prefix,
+                    // find a solution how to wrap the field inside custom Field (without owner?)
+                    $modelCloned = clone $model;
+                    $modelCloned->persistenceData['use_table_prefixes'] = true;
+
+                    $arguments[$identifier] = $modelCloned->getField($identifier);
```

Comment:
All fields needs to be aliased when CTE is used too - https://dbfiddle.uk/R7lgFv8w.